Smart layout API (#671)
sketch-hq/SketchAPI: docs/api/SmartLayout.md

---
title: SmartLayout
order: 215
section: models
---

> The [SmartLayout](#smartlayout) object contains the set of possible smart layouts that can be applied to [SymbolMaster](#symbol-master) and [Group](#group) layers.

Reviewer: ```suggestion
The [SmartLayout](#smartlayout) object contains the set of possible Smart Layouts that can be applied to [SymbolMaster](#symbol-master) and [Group](#group) layers.
```

| Properties         |                                                     |
| ------------------ | --------------------------------------------------- |
| LeftToRight        | Smart layout flowing left to right                  |
| HorizontallyCenter | Smart layout expanding horizontally from the center |
| RightToLeft        | Smart layout flowing right to left                  |
| TopToBottom        | Smart layout flowing from top to bottom             |
| VerticallyCenter   | Smart layout expanding verically from the center    |

> | BottomToTop        | Smart layout flowing from bottom to top             |

Reviewer: Replace with _Smart Layout_ for consistency.

> ## Set a smart layout

Reviewer: ```suggestion
## Set a Smart Layout
```

```javascript
const SmartLayout = require('sketch').SmartLayout
layer.smartLayout = SmartLayout.TopToBottom
```

> Given a reference to a [SymbolMaster](#symbol-master) or [Group](#group) layer use the `smartLayout` setter to apply one of the smart layout values.

Reviewer: ```suggestion
Given a reference to a [SymbolMaster](#symbol-master) or [Group](#group) layer use the `smartLayout` setter to apply one of the Smart Layout values.
```

> ## Clear a smart layout

Reviewer: ```suggestion
## Clear a Smart Layout
```

> Set the `smartLayout` value to `null` to remove the smart layout. This is the equivalent of selecting "None" in the UI.

Reviewer: ```suggestion
Set the `smartLayout` value to `null` to remove the Smart Layout. This is the equivalent of selecting "None" in the Sketch Inspector.
```

Maybe also worth clarifying that the dimensions remain the same after clearing the layout?

```javascript
symbolMaster.smartLayout = null
```

## Trigger a smart layout

```javascript
const SmartLayout = require('sketch').SmartLayout
symbolMaster.smartLayout = SmartLayout.TopToBottom
symbolInstance.resizeWithSmartLayout()
```

> In order to trigger a smart layout resize in an instance, for example after changing an override value, call the `resizeWithSmartLayout()` method on the [SymbolInstance](#symbol-instance) layer.

Reviewer: ```suggestion
In order to trigger a Smart Layout resize in an instance, for example after changing an override value, call the `resizeWithSmartLayout()` method on the [SymbolInstance](#symbol-instance) layer.
```
